Order return of collections.list_all() alphabetically by key

saireddythfc opened the pull request:
1. Modified the _collection_configs_from_json and _collection_configs_simple_from_json functions in weaviate/collections/classes/config_methods.py to sort the dictionaries by key before returning them.

2. Added unit tests in test/collection/test_collections_sorting.py to verify the sorting functionality.

3. Added integration tests in integration/test_collections_sorting.py to verify the sorting in a real environment.

saireddythfc commented:
1. Modified the _collection_configs_from_json and _collection_configs_simple_from_json functions in weaviate/collections/classes/config_methods.py to sort the dictionaries by key before returning them.

2. Added unit tests in test/collection/test_collections_sorting.py to verify the sorting functionality.

3. Added integration tests in integration/test_collections_sorting.py to verify the sorting in a real environment.

tsmith023 commented:
Hi @saireddythfc, thanks for the contribution! Looks like the linter needs to be ran on your code, we use `black` and `flake8`, plus the test you've introduced is not scoped for `pytest` appropriately. Otherwise, the changes look good! Once you fix the tests, I'll give the PR a formal review 😁

saireddythfc commented:
Update:

1. Fixed name collision occurring in integration test (client -> sorting_test_client).
2. Changed pytest fixture from module to function.
3. Manually ran black linter before pre-commit did its task on commit.

tsmith023 commented on integration/test_collections_sorting.py:
```diff
@@ -0,0 +1,41 @@
+from integration.conftest import ClientFactory
+
+
+@pytest.fixture
+def sorting_test_client(client_factory: ClientFactory):
+    client = client_factory()
+    client.collections.delete_all()
+    yield client
+    client.collections.delete_all()
+    client.close()
```
I like the idea of making a specific collection but the `client.collections.delete_all()` calls are likely to cause data race issues with other concurrent tests that run in the CI pipeline. I think is perhaps the origin of the flakes within the CI runs of this PR

In `integration.conftest`, there is the `CollectionFactory`. Can you use that here instead? It handles automatically creating collections for you with unique names and subsequently deleting them when the test finishes. wdyt?

tsmith023 reviewed (changes requested):
Implementation LGTM, just the test needs fixing imo!

tsmith023 reviewed (approved):
LGTM now 😁 The only thing left is the linter complaining about a docstring!
> integration/test_collections_sorting.py:15:1: D202 No blank lines allowed after function docstring

Once fixed, I'll merge

saireddythfc commented:
I agree to the CLA shown by @weaviate-git-bot
